utilities/rest-helper-factory.js

```diff
@@ -585,6 +590,285 @@ module.exports = function(logger, mongoose, server) {
+     * @param server: A Hapi server.
+     * @param model: A mongoose model.
+     * @param options: Options object.
+     * @param logger: A logging object.
+     */
+    // TODO: handle partial recovers (return list of ids that failed/were not found)
+    generateRecoverManyEndpoint: function(server, model, options, logger) {
+      // This line must come first
+      validationHelper.validateModel(model, logger)
+      const Log = logger.bind(chalk.yellow('RecoverMany'))
+
+      const collectionName = model.collectionDisplayName || model.modelName
+      if (config.logRoutes) {
+        Log.note('Generating Recover Many endpoint for ' + collectionName)
+      }
+
+      options = options || {}
+
+      let resourceAliasForRoute
+
+      if (model.routeOptions) {
+        resourceAliasForRoute = model.routeOptions.alias || model.modelName
+      } else {
+        resourceAliasForRoute = model.modelName
+      }
+
+      const handler = HandlerHelper.generateRecoverHandler(model, options, Log)
+
+      let payloadModel = null
+      /* if (config.enableSoftDelete) {
+        payloadModel = Joi.alternatives().try(
+          Joi.array().items(
+            Joi.object({
+              _id: Joi.objectId()
+            })
+          ),
+          Joi.array().items(Joi.objectId())
+        )
+      } else {
+        payloadModel = Joi.array().items(Joi.objectId())
+      } */
+
+      payloadModel = Joi.alternatives().try(
+        Joi.array().items(
+          Joi.object({
+            _id: Joi.objectId()
+          })
+        ),
+        Joi.array().items(Joi.objectId())
+      )
+
+      if (!config.enablePayloadValidation) {
+        payloadModel = Joi.alternatives().try(payloadModel, Joi.any())
+      }
+
+      let auth = false
+      let recoverManyHeadersValidation = Object.assign(headersValidation, {})
+
+      if (config.authStrategy && model.routeOptions.recoverAuth !== false) {
+        auth = {
+          strategy: config.authStrategy
+        }
+
+        const scope = authHelper.generateScopeForEndpoint(model, 'recover', Log)
+
+        if (!_.isEmpty(scope)) {
+          auth.scope = scope
+          if (config.logScopes) {
+            Log.debug('Scope for PUT/' + resourceAliasForRoute + ':', scope)
+          }
+        }
+      } else {
+        recoverManyHeadersValidation = null
+      }
+
+      let policies = []
+
+      if (model.routeOptions.policies && config.enablePolicies) {
+        policies = model.routeOptions.policies
+        policies = (policies.rootPolicies || []).concat(
+          policies.recoverPolicies || []
+        )
+      }
+
+      if (config.enableDocumentScopes && auth) {
+        policies.push(restHapiPolicies.enforceDocumentScopePre(model, Log))
+        policies.push(restHapiPolicies.enforceDocumentScopePost(model, Log))
+      }
+
+      /* if (config.enableDeletedBy && config.enableSoftDelete) {
+        policies.push(restHapiPolicies.addDeletedBy(model, Log))
+      } */
```

Comment:
Since this is an update action, let's use:

```
if (config.enableUpdatedBy) {
        policies.push(restHapiPolicies.addUpdatedBy(model, Log))
}
```